src/mappings/entities/mapToken.ts

```diff
@@ -0,0 +1,165 @@
+import BN from 'bn.js';
+import { SubstrateExtrinsic } from '@subql/types';
+import { ModuleIdEnum, CallIdEnum } from './common';
+import { Token } from '../../types';
+
+const chainNumberMultiplier = new BN(1000000);
+
+export async function mapToken(
+  blockId: number,
+  callId: CallIdEnum,
+  moduleId: ModuleIdEnum,
+  params: Record<string, any>,
+  extrinsic: SubstrateExtrinsic,
+): Promise<void> {
+  if (!extrinsic.success || ![ModuleIdEnum.Asset].includes(moduleId)) {
+    return;
+  }
+
+  if (callId === CallIdEnum.CreateAsset) {
+    const {
+      name,
+      ticker,
+      assetType: type,
+      fundingRound,
+      divisible,
+      disableIu,
+      identifiers,
+    } = params;
+    const ownerDid = extrinsic.events[1].event.data[0].toString();
+    await Token.create({
+      id: ticker,
+      ticker,
+      name,
+      type,
+      fundingRound,
+      isDivisible: divisible,
+      isFrozen: false,
+      isUniquenessRequired: !disableIu,
+      documents: [],
+      identifiers: JSON.stringify(identifiers),
+      ownerDid,
+      fullAgents: [ownerDid],
+      holders: [],
+      totalSupply: '0',
+    }).save();
+  }
+
+  if (callId === CallIdEnum.RenameAsset) {
+    const { ticker, name: newName } = params;
+    const token = await Token.getByTicker(ticker);
+    if (!token) return;
```

Comment:
If this happens then the DB is in a bad state. We should throw an error in this case.

Reply:
True. I added it as fail safe, wasn't sure if it'll break the whole thing if it happens. We could implement error reporting behind the scenes for some thing like this. Or is it fine to throw an error?

Reply:
I think its fine to throw an error. It will stop indexing / break it, but I think its better than showing wrong or missing data.

Considering how this gets deployed, we should still have the old version up

I am pretty sure we throw errors in the native branch so it should be alright.